framework/tlvf/AutoGenerated/src/tlvf/ieee_1905_1/tlvDeviceInformation.cpp

```diff
@@ -37,59 +37,83 @@ sMacAddr& tlvDeviceInformation::mac() {
+uint8_t& tlvDeviceInformation::local_interface_list_length() {
+    return (uint8_t&)(*m_local_interface_list_length);
 }
 
-std::tuple<bool, tlvDeviceInformation::sInfo&> tlvDeviceInformation::info(size_t idx) {
-    bool ret_success = ( (m_info_idx__ > 0) && (m_info_idx__ > idx) );
+std::tuple<bool, cLocalInterfaceInfo&> tlvDeviceInformation::local_interface_list(size_t idx) {
+    bool ret_success = ( (m_local_interface_list_idx__ > 0) && (m_local_interface_list_idx__ > idx) );
     size_t ret_idx = ret_success ? idx : 0;
     if (!ret_success) {
         TLVF_LOG(ERROR) << "Requested index is greater than the number of available entries";
     }
-    return std::forward_as_tuple(ret_success, m_info[ret_idx]);
+    return std::forward_as_tuple(ret_success, *(m_local_interface_list_vector[ret_idx]));
 }
 
-bool tlvDeviceInformation::alloc_info(size_t count) {
-    if (m_lock_order_counter__ > 0) {;
-        TLVF_LOG(ERROR) << "Out of order allocation for variable length list info, abort!";
-        return false;
+std::shared_ptr<cLocalInterfaceInfo> tlvDeviceInformation::create_local_interface_list() {
+    if (m_lock_order_counter__ > 0) {
+        TLVF_LOG(ERROR) << "Out of order allocation for variable length list local_interface_list, abort!";
+        return nullptr;
     }
-    if (count == 0) {
-        TLVF_LOG(WARNING) << "can't allocate 0 bytes";
-        return false;
-    }
-    size_t len = sizeof(sInfo) * count;
-    if(getBuffRemainingBytes() < len )  {
-        TLVF_LOG(ERROR) << "Not enough available space on buffer - can't allocate";
-        return false;
+    size_t len = cLocalInterfaceInfo::get_initial_size();
+    if (m_lock_allocation__ || getBuffRemainingBytes() < len) {
+        TLVF_LOG(ERROR) << "Not enough available space on buffer";
+        return nullptr;
     }
     m_lock_order_counter__ = 0;
-    uint8_t *src = (uint8_t *)&m_info[*m_info_length];
-    uint8_t *dst = src + len;
+    m_lock_allocation__ = true;
+    uint8_t *src = (uint8_t *)m_local_interface_list;
+    if (m_local_interface_list_idx__ > 0) {
+        src = (uint8_t *)m_local_interface_list_vector[m_local_interface_list_idx__ - 1]->getBuffPtr();
+    }
     if (!m_parse__) {
+        uint8_t *dst = src + len;
         size_t move_length = getBuffRemainingBytes(src) - len;
         std::copy_n(src, move_length, dst);
     }
-    m_info_idx__ += count;
-    *m_info_length += count;
+    return std::make_shared<cLocalInterfaceInfo>(src, getBuffRemainingBytes(src), m_parse__);
+}
+
+bool tlvDeviceInformation::add_local_interface_list(std::shared_ptr<cLocalInterfaceInfo> ptr) {
+    if (ptr == nullptr) {
+        TLVF_LOG(ERROR) << "Received entry is nullptr";
+        return false;
+    }
+    if (m_lock_allocation__ == false) {
+        TLVF_LOG(ERROR) << "No call to create_local_interface_list was called before add_local_interface_list";
+        return false;
+    }
+    uint8_t *src = (uint8_t *)m_local_interface_list;
```

Comment:
Is this cast really necessary? If it is, use `reinterpret_cast` (I think)

Reply:
This is auto-generated code